packages/teams-js/src/public/video.ts

```diff
@@ -216,4 +216,92 @@ export namespace video {
   export function isSupported(): boolean {
     return ensureInitialized(runtime) && runtime.supports.video ? true : false;
   }
+
+  /**
+   * Namespace to get video frames from a media stream
+   */
+  export namespace mediaStream {
+    export function isSupported(): boolean {
+      return ensureInitialized(runtime) && textureStreamAvailable() && !!runtime.supports.video?.mediaStream;
+    }
+
+    function textureStreamAvailable(): boolean {
+      return !!(window['chrome']?.webview?.getTextureStream && window['chrome']?.webview?.registerTextureStream);
+    }
+
+    export type ReceivedVideoFrame = {
+      videoFrame: VideoFrame;
+    };
+
+    /**
+     * Video effect change call back function definition.
+     * The video app should resolve the promise to notify a successfully processed video frame.
+     * The video app should reject the promise to notify a failure.
+     */
+    export type VideoFrameCallback = (receivedVideoFrame: ReceivedVideoFrame) => Promise<VideoFrame>;
```

Comment:
It seems that we can receive other types, such as CanvasImageSource? If so, we can support more return types